driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala

```diff
@@ -166,8 +228,12 @@ class Driver private(driverConf: DriverConf,
 
   // TODO remove infinite duration
   // TODO remove askPattern
-  private def askCommand[T <: ServerReply : ClassTag](remoteActor: ActorRef, message: Any, timeout: FiniteDuration = 1 day): Future[T] = {
-    remoteActor.ask(message)(timeout).mapTo[T]
+  private def askCommand(remoteActor: ActorRef,
+                         message: CommandEnvelope,
```

Comment:
Maybe:
private def askCommand(command: CommandEnvelope)(implicit val remoteActor: ActorRef) and replace val proxy actor with implicit val proxyActor or even use it directly.